### DISCUSSION TITLE
Load script tag from CRM

### DISCUSSION THREAD

**Discussion by YuLogun** [nuxt/nuxt#17077] [Questions] [answered]
Hi! 

There are quite a lot of articles regarding how to include script tags in Nuxt (using plugins, nuxt.config or Vue-meta).

However, all of them require the developer to configure the script options manually (src, defer, etc). 
But it doesn’t take into account cases when the developer doesn’t in advance know the content of the tag that should be loaded. 

My personal case:
At our project we have an admin page and Backend in PHP and our Frontend is using Nuxt 3.
At that admin page people can enter different information which we will then get on the Frontend from API. One of the things that administrators can add is adding different metrics scripts or third party scripts. 
For example, we use Bank131 widget to make payments. Code and styles for this widget differ for dev and pros environments.  Here’s the example:

    ```<link
      href="https://widget.bank131.ru/card-tokenizer.css"
      rel="stylesheet"
    />
    <script src="https://widget.bank131.ru/card-tokenizer.js" defer></script>```

On the Frontend I’m using an API endpoint to fetch these tags and then I add them into body. 
However, what if I need these tags only on payment page? 
If i navigate to this page via client side navigation the script above is not being loaded. It loads only after refreshing the page. 

What’s the solution to this issue?
Filling in the scripts in the admin page and getting it from API is the constraint I must follow.

**Comment by huang-julien** [comment]
hi, you can add these tags using useHead 
https://v3.nuxtjs.org/guide/features/head-management/
retrieve your scripts from your api then pass them into `useHead`
````ts
const data = {
  script: [{
  src: 'https://widget.bank131.ru/card-tokenizer.js',
  defer: true
 }]
} // retrieved from your api

usehead(data)
````

**Comment by YuLogun** [comment]
Hi! 
thanks for your time and reply!

The problem is, as I stated, I get the tags from an API endpoint. The response is a string
`"<link href=\"https://widget.bank131.ru/payment-form.css\" rel=\"stylesheet\" />\r\n<script src=\"https://widget.bank131.ru/payment-form.js\" defer></script>"`
(which is why I cannot use `useHead`).

Then I'm using this component to render them
```
<script lang="ts" setup>
const props = defineProps<{
    script: string
}>();

const ScriptTag = () => {
    return h('div', {
        innerHTML: props.script || ''
    });
};
</script>

<template>
    <ScriptTag />
</template>
```

They do appear in the DOM, however the script only loads if I refresh the page. If I navigate to it from another Nuxt page, I get an error that Bank131 is not defined (thus its script is not loaded).

**Comment by huang-julien** [comment, marked as answer]
When you refresh the page, the script is executed because your browser receive the html with the script tags. However, injecting scripts using innerHTML does not work, that's why you don't get your script executed when navigating through pages once in client side.

Did you try to use DOMParser to parse your tags ? You would be able to transform it into an object to send it to `useHead`.

**Reply by YuLogun** [reply]
Thanks a lot! What a great solution!
